### PR TITLE
Only load the overlapping dbSNP sites when using target intervals in

### PR DESCRIPTION
**Pull request by nh13**
CollectVariantCallingMetrics.  This should speed up the load
significantly for smaller regions.

@tfenne @yfarjoun 

I tested with an interval list with only a few hundred intervals.  It took only a few seconds, versus almost a full minute previously.  I used the dsnp file found here: ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/2.8/b37/dbsnp_138.b37.excluding_sites_after_129.vcf.gz

[Review on Reviewable](https://reviewable.io/reviews/broadinstitute/picard/429)

### CODE REVIEW

File: src/java/picard/util/DbSnpBitSetUtil.java [hunk 1]
```diff
@@ -66,39 +76,56 @@ public DbSnpBitSetUtil(final File dbSnpFile, final SAMSequenceDictionary sequenc
...
-                                final Map<DbSnpBitSetUtil, Set<VariantType>> bitSetsToVariantTypes) {
+                                final Map<DbSnpBitSetUtil, Set<VariantType>> bitSetsToVariantTypes,
+                                final IntervalList intervals) {
 
-        final VCFFileReader variantReader = new VCFFileReader(dbSnpFile);
-        final CloseableIterator<VariantContext> variantIterator = variantReader.iterator();
+        final VCFFileReader variantReader = new VCFFileReader(dbSnpFile, intervals != null);
+        final Iterator<VariantContext> variantIterator;
+        if (intervals != null) {
+            variantIterator = new ByIntervalListVariantContextIterator(variantReader, intervals);
```
Comment (tfenne) [inline src/java/picard/util/DbSnpBitSetUtil.java]: I wish I had known this existed previously!  Why isn't there are VCFFileReader.query(intervals) that creates one of these?  Sigh.

**Comment by tfenne** [comment]
Looks good to me @nh13.  Probably @yfarjoun will want to take a look too?

File: src/java/picard/util/DbSnpBitSetUtil.java (hunk 1 above)
Comment (nh13) [inline src/java/picard/util/DbSnpBitSetUtil.java]: You're the author on `ByIntervalListVariantContextIterator`!

Comment (tfenne) [inline src/java/picard/util/DbSnpBitSetUtil.java]: :frowning:

**Comment by tfenne** [comment]
@yfarjoun I'd like to merge this PR as not having it is annoying.  Any objections?

**Comment by yfarjoun** [comment]
Sorry for the delay. Thanks for being considerate of my dalliance with the
Florida sun. I'll take a look now.

File: src/java/picard/util/DbSnpBitSetUtil.java [hunk 2]
```diff
@@ -24,13 +24,16 @@
...
-import htsjdk.samtools.util.CloseableIterator;
 import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.IntervalList;
 import htsjdk.variant.variantcontext.VariantContext;
 import htsjdk.variant.vcf.VCFFileReader;
+import picard.vcf.ByIntervalListVariantContextIterator;
 
 import java.io.File;
-import java.lang.IllegalArgumentException;import java.lang.String;import java.util.BitSet;import java.util.Collection;import java.util.EnumSet;import java.util.HashMap;import java.util.Map;import java.util.Set;
+import java.lang.IllegalArgumentException;import java.lang.String;import java.util.BitSet;import java.util.Collection;import java.util.EnumSet;import java.util.HashMap;
```
Comment (yfarjoun) [inline src/java/picard/util/DbSnpBitSetUtil.java]: Why do these import statements all fall on the same line? can you fix that? I know you didn't do it initially.

**Comment by yfarjoun** [comment]
:+1: please fix import lines and rebase, as there have been a few merges since yours. 

Thanks!

**Comment by nh13** [comment]
@yfarjoun @tfenne : imports fixed and rebased.

**Comment by nh13** [comment]
@vdauwera after rebase, it looks like travis is failing (even master is failing).  Could you take a look?

**Pull request merged**